Implementation smell,Namespace,Class,File,Method,Description
Empty Catch Block,ExceptionFilters,DirectThrowingFilter,C:\repos\jskeet_DemoCode\Abusing CSharp\Code\ExceptionFilters\DirectThrowingFilter.cs,Foo,The method has an empty catch block.
Empty Catch Block,ExceptionFilters,ExceptionContext,C:\repos\jskeet_DemoCode\Abusing CSharp\Code\ExceptionFilters\ExceptionContext.cs,OuterMethod,The method has an empty catch block.
Empty Catch Block,ExceptionFilters,ExceptionContext,C:\repos\jskeet_DemoCode\Abusing CSharp\Code\ExceptionFilters\ExceptionContext.cs,InnerMethod,The method has an empty catch block.
Empty Catch Block,ExceptionFilters,ReflectionThrowingFilter,C:\repos\jskeet_DemoCode\Abusing CSharp\Code\ExceptionFilters\ReflectionThrowingFilter.cs,Foo,The method has an empty catch block.
Magic Number,ExceptionFilters,DirectThrowingFilter,C:\repos\jskeet_DemoCode\Abusing CSharp\Code\ExceptionFilters\DirectThrowingFilter.cs,Foo,The following statement contains a magic number: try              {                  Bar();              }              catch (Exception e) when (e.Message[10] == '!')              {              }
Magic Number,ExceptionFilters,ExceptionContext,C:\repos\jskeet_DemoCode\Abusing CSharp\Code\ExceptionFilters\ExceptionContext.cs,Main,The following statement contains a magic number: try              {                  OuterMethod(10);              }              catch (IOException e)              {                  Console.WriteLine("Exception: {0}"' e);                  Console.WriteLine();                  e.DumpContext();              }
Magic Number,ExceptionFilters,ExceptionContext,C:\repos\jskeet_DemoCode\Abusing CSharp\Code\ExceptionFilters\ExceptionContext.cs,InnerMethod,The following statement contains a magic number: try              {                  if (x == 7)                  {                      GoBang();                  }              }              catch (Exception e) when (new LogContextEntry {[nameof(x)] = x'["now"] = DateTime.Now }.AddTo(e))              { /* Never get here... */ }
Magic Number,ExceptionFilters,ReflectionThrowingFilter,C:\repos\jskeet_DemoCode\Abusing CSharp\Code\ExceptionFilters\ReflectionThrowingFilter.cs,Foo,The following statement contains a magic number: try              {                  Bar();              }              catch (Exception e) when (e.Message[10] == '!')              {              }
